### PR TITLE
Added an argument for different blacklisted url handling

### PR DESCRIPTION
**Pull request by guykeller**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23344

## Description
Until now, if any url in your list was blacklisted, an error would have been returned.
That will now depend on an argument

## Does it break backward compatibility?
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Integrations/UrlScan/UrlScan.py [hunk 1]
```diff
@@ -46,13 +48,24 @@ def http_request(method, url_suffix, json=None, wait=0, retries=0):
...
+
         if r.status_code == 429:
             if retries <= 0:
                 # Error in API call to URLScan.io [429] - Too Many Requests
                 return_error('API rate limit reached. Use the retries and wait arguments when submitting multiple URls')
             else:
                 time.sleep(wait)
                 return http_request(method, url_suffix, json, wait, retries - 1)
+        response_json = r.json()
+        # raise ValueError(f'Response: {response_json}')
```
Comment (ronykoz) [inline Integrations/UrlScan/UrlScan.py:60]: remove

File: Integrations/UrlScan/UrlScan.py [hunk 2]
```diff
@@ -46,13 +48,24 @@ def http_request(method, url_suffix, json=None, wait=0, retries=0):
...
+        error_description = response_json.get('description')
+        should_continue_on_blacklisted_urls = demisto.args().get('continue_on_blacklisted_urls')
+        if should_continue_on_blacklisted_urls and error_description == BLACKLISTED_URL_ERROR_MESSAGE:
+            response_json['uuid'] = -1 # TODO : check what happens with that uuid and how that could be breaking
+            requested_url = JSON.loads(json)['url']
+            demisto.results(f'The URL {requested_url} is blacklisted, thus no results will be returned for it.')
```
Comment (ronykoz) [inline Integrations/UrlScan/UrlScan.py:66]: maybe return some context info about this?

File: Integrations/UrlScan/UrlScan.py [hunk 3]
```diff
@@ -46,13 +48,24 @@ def http_request(method, url_suffix, json=None, wait=0, retries=0):
...
                 return_error('API rate limit reached. Use the retries and wait arguments when submitting multiple URls')
             else:
                 time.sleep(wait)
                 return http_request(method, url_suffix, json, wait, retries - 1)
+        response_json = r.json()
+        # raise ValueError(f'Response: {response_json}')
+        error_description = response_json.get('description')
+        should_continue_on_blacklisted_urls = demisto.args().get('continue_on_blacklisted_urls')
+        if should_continue_on_blacklisted_urls and error_description == BLACKLISTED_URL_ERROR_MESSAGE:
+            response_json['uuid'] = -1 # TODO : check what happens with that uuid and how that could be breaking
```
Comment (ronykoz) [inline Integrations/UrlScan/UrlScan.py:64]: add a new key to the dict instead, and then use get with default value

File: Integrations/UrlScan/UrlScan.py (hunk 2 above)
Comment (guykeller) [reply Integrations/UrlScan/UrlScan.py:65]: We can add a new context path, I don't think we can update any of the existing ones

File: Integrations/UrlScan/UrlScan.py [hunk 4]
```diff
@@ -158,8 +170,9 @@ def urlscan_submit_url():
     wait = int(demisto.args().get('wait', 5))
     retries = int(demisto.args().get('retries', 0))
     r = http_request('POST', 'scan/', sub_json, wait, retries)
-    uuid = r['uuid']
-    return uuid
+    return r
+    #uuid = r['uuid']
+    #return uuid
```
Comment (ronykoz) [inline Integrations/UrlScan/UrlScan.py:175]: delete

**Comment by roysagi** [comment]
RN looks good

**Comment by guykeller** [comment]
Opened a new PR, closing this

**Pull request closed without merging**